airbyte-webapp/src/pages/connectorBuilder/ConnectorBuilderLandingPage/ConnectorBuilderLandingPage.tsx

```diff
@@ -0,0 +1,214 @@
+import { ReactComponent as ImportYamlImage } from "../../../../public/images/connector-builder/import-yaml.svg";
+import { ReactComponent as StartFromScratchImage } from "../../../../public/images/connector-builder/start-from-scratch.svg";
+import { ConnectorBuilderRoutePaths } from "../ConnectorBuilderRoutes";
+
+const YAML_UPLOAD_ERROR_ID = "connectorBuilder.yamlUpload.error";
+
+const ConnectorBuilderLandingPageInner: React.FC = () => {
+  const { storedFormValues, setStoredFormValues, storedManifest, setStoredManifest, setStoredEditorView } =
+    useConnectorBuilderLocalStorage();
+  const navigate = useNavigate();
+
+  // use refs for the intial values because useLocalStorage changes the references on re-render
+  const initialStoredFormValues = useRef<BuilderFormValues>(storedFormValues);
+  const initialStoredManifest = useRef<ConnectorManifest>(storedManifest);
+
+  useEffect(() => {
+    if (
+      !isEqual(initialStoredFormValues.current, DEFAULT_BUILDER_FORM_VALUES) ||
+      !isEqual(initialStoredManifest.current, DEFAULT_JSON_MANIFEST_VALUES)
+    ) {
+      navigate(ConnectorBuilderRoutePaths.Edit);
+    }
+  }, [navigate]);
+
+  const fileInputRef = useRef<HTMLInputElement>(null);
+  const { registerNotification, unregisterNotificationById } = useNotificationService();
+  const { convertToBuilderFormValues } = useManifestToBuilderForm();
+  const [importYamlLoading, setImportYamlLoading] = useState(false);
+
+  const handleYamlUpload = useCallback(
+    async (uploadEvent: React.ChangeEvent<HTMLInputElement>) => {
+      setImportYamlLoading(true);
+      const file = uploadEvent.target.files?.[0];
+      const reader = new FileReader();
+      reader.onload = async (readerEvent) => {
+        const yaml = readerEvent.target?.result as string;
+        const fileName = file?.name;
+
+        try {
+          let json;
+          try {
+            json = load(yaml) as ConnectorManifest;
+          } catch (e) {
+            if (e instanceof YAMLException) {
+              registerNotification({
+                id: YAML_UPLOAD_ERROR_ID,
+                text: (
+                  <FormattedMessage
+                    id={YAML_UPLOAD_ERROR_ID}
+                    values={{
+                      reason: e.reason,
+                      line: e.mark.line,
+                    }}
+                  />
+                ),
+                type: ToastType.ERROR,
+              });
+            }
+            return;
+          }
+
+          let convertedFormValues;
+          try {
+            convertedFormValues = await convertToBuilderFormValues(json, DEFAULT_BUILDER_FORM_VALUES);
+          } catch (e) {
+            setStoredEditorView("yaml");
+            setStoredManifest(json);
+            navigate(ConnectorBuilderRoutePaths.Edit);
+            return;
+          }
+
+          if (fileName) {
+            convertedFormValues.global.connectorName = startCase(fileName.split(".")[0]);
```

Comment:
This used to work really well until https://github.com/airbytehq/airbyte/pull/21771 - now it will default to "Manifest" which is not super helpful. I think the next best thing is the url base, but no strong opinion on this.

My suggestion:
* If the file name is not "manifest", use the current logic
* If we can turn into form values, use the base url